### ISSUE TITLE
Extend Envoy to get custom metrics needed to implement smi metrics

### ISSUE THREAD

**Issue by michelleN** [openservicemesh/osm#984]

**Comment by nojnhuh** [comment]
This is largely complete, but may need small changes as we continue with the SMI metrics implementation.

**Comment by nojnhuh** [comment]
Just remembered @eduser25 is still working on the request duration metrics. Reopening to finish tracking that.

**Comment by nojnhuh** [comment]
Now this is basically complete. Only waiting on upstream Envoy to fully support WASM filters before merging. All code is on my fork: https://github.com/nojnhuh/osm/tree/envoy-wasm

**Comment by nojnhuh** [comment]
Reopening until this is merged.

**Comment by michelleN** [comment]
Now that the wasm work has been merged into Envoy, the next steps should be:
1. Wait for Envoy 1.17 release 
2. Update Envoy to 1.17 in osm
3. merge the wasm filter work @nojnhuh has done into osm
4. Cut a release
5. Either create a new repo under openservicemesh to host the smi-metrics implementation OR merge smi-metrics implementation into the servicemeshinterface/smi-metrics repo (or both)
6. update docs

probably need to discuss step 3 a little bit more. What do u think about these next steps ^ @nojnhuh ?

**Comment by nojnhuh** [comment]
@michelleN That's what I was thinking of at a high level. I agree that we should think through how we want to merge everything. I was planning on splitting my OSM branch into a reasonable number of smaller commits and making each one of those a PR to the main branch. There's a lot of junk commits on my branch currently from when I was still trying to figure things out. I haven't gone through the exercise of defining what each of those PRs should look like yet.

Side note: Since v1.17 of Envoy isn't going to be released until well after our v0.6.0, should we clear the milestone on this for now?

**Comment by nojnhuh** [comment]
Taking this off the v0.6.0 milestone since WASM will likely not GA in Envoy until early 2021 with v1.17.0.

**Comment by nojnhuh** [comment]
DONE with #2616 :tada:
